### PR TITLE
feat-added wishlist functionality

### PR DESCRIPTION
**Pull request by aryan-13**
Added the following:
- Wishlist context
- add to wishlist functionality
- delete items from wishslist

### CODE REVIEW

**Review by rehanmohiuddin — APPROVED** [review]

File: src/Components/ProductCard.jsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+import { Link } from 'react-router-dom';
 import { useCart } from '../Context/cart-context';
+import { useWishlist } from '../Context/wishlist-context';
 
-function ProductCard({ product }) {
+function ProductCard({ product, page }) {
 	const { state, dispatch } = useCart();
+	const { wishlistDispatch } = useWishlist();
 	const checkInCart = () => {
 		for (let i = 0; i < state.cartProducts.length; i++) {
```
Comment (saadahmadhere) [inline src/Components/ProductCard.jsx]: you could've used higher order function here.

File: src/Context/wishlist-context.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { useContext, createContext, useReducer } from 'react';
+// import { useCart } from '../Context/cart-context';
```
Comment (saadahmadhere) [inline src/Context/wishlist-context.jsx]: unwanted comments should be removed

File: src/Context/wishlist-context.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+function WishlistProvider({ children }) {
+	const addToWishlist = (state, product) => {
+		if (state.some((item) => item.id === product.id)) return state;
+		else return [...state, product];
+	};
+	const removeFromWishlist = (state, product) => {
+		return state.filter((item) => item.id !== product.id);
+	};
+	// const moveToWishlist = (state, product) => {
```
Comment (saadahmadhere) [inline src/Context/wishlist-context.jsx]: unwanted comments should be removed.

File: src/pages/ProductListing.jsx [hunk 4]
```diff
@@ -1,39 +1,42 @@
...
-          {products.map(product => {
-            return (
-              <ProductCard product={product} key={product.id}/>
-            )
-          })}
-        </div>
-      </div>
-    </div>
-  )
+	const [products, setProducts] = React.useState([]);
```
Comment (saadahmadhere) [inline src/pages/ProductListing.jsx:8]: as in line number 2 (with useEffect), you could've imported useState() from react (destructured it) and used it here directly.

File: src/pages/WishlistPage.jsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
```
Comment (saadahmadhere) [inline src/pages/WishlistPage.jsx]: No need to import React in the newer versions.

**Review by saadahmadhere — COMMENTED** [review]
Nice use of context to avoid prop drilling. Made some suggestions, otherwise the code looks fine.

File: src/Components/ProductCard.jsx (hunk 1 above)
Comment (aryan-13) [reply src/Components/ProductCard.jsx]: Yes, Changed it!

**Comment by aryan-13** [comment]
Made the suggested changes!

**Review by anshikaa12 — APPROVED** [review]
code is well structured and good to merge

**Pull request merged**